### ISSUE TITLE
MemberExtension::updateValidator() hook doesn’t work when editing via Groups

### ISSUE THREAD

**Issue by kinglozzer** [silverstripe/silverstripe-framework#9184]
## Affected Version

4.4, possibly earlier versions

## Description

We provide a hook to customise `Member_Validator` in extensions: https://github.com/silverstripe/silverstripe-framework/blob/4.4.3/src/Security/Member.php#L764.

When editing a group, there’s a GridField of members which you can use to edit members within that group. In this view, the above extension hook is never called because it’s set as a validator like so:

https://github.com/silverstripe/silverstripe-framework/blob/4.4.3/src/Security/Group.php#L167-L168

**Comment by kinglozzer** [comment]
Simplest solution seems to be to move the logic for adding a validator into the `setItemEditFormCallback()` closure:

```php
->setItemEditFormCallback(function ($form) use ($group) {
    $record = $form->getRecord();
    $form->setValidator($record->getValidator());
    // ... etc
```

**Comment by ScopeyNZ** [comment]
Closed by #9255
